OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -195,8 +208,30 @@ public void Tick(World world)
 			if (info.ContrailLength > 0)
 				contrail.Update(pos);
 
+			var flightLengthReached = ticks++ >= length;
+			var shouldBounce = remainingBounces > 0;
+
+			if (flightLengthReached && shouldBounce)
+			{
+				shouldExplode |= AnyValidTargetsInRadius(world, pos, info.Width + info.TargetExtraSearchRadius, args.SourceActor, true);
+				target += (pos - source) * info.BounceRangeModifier / 100;
+				var dat = world.Map.DistanceAboveTerrain(target);
+				target += new WVec(0, 0, -dat.Length);
+				length = Math.Max((target - pos).Length / speed.Length, 1);
+				ticks = 0;
+				source = pos;
+				remainingBounces--;
+			}
+
 			// Flight length reached / exceeded
-			shouldExplode |= ticks++ >= length;
+			shouldExplode |= flightLengthReached && !shouldBounce;
+
+			// Driving into cell with higher height level
+			shouldExplode |= world.Map.DistanceAboveTerrain(pos).Length < 0;
```

Comment:
Honestly, I liked how old Bullet behaviour even ignored cliffs as well, so I planned to open a PR with this check being exposed into yaml, if taht's fine by everyone.